worker/changestream/stream/stream.go

```diff
@@ -128,21 +128,26 @@ func (s *Stream) loop() error {
 					if !ok || !inner {
 						break INNER
 					}
+					// If we get a fileCreated event, and we're already waiting
+					// for a fileRemoved event, then we're in the middle of a
+					// block, so just continue.
 					s.logger.Debugf("ignoring file change event")
 				}
 			}
 
 			s.logger.Infof("Change stream has been unblocked")
 
 		case <-timer.Chan():
-			changes, err := s.readChanges()
+			// We pass this context to every database method that accepts one.
+			// It is cancelled by killing the tomb, which prevents shutdown
+			// being blocked by such calls.
+			ctx := s.tomb.Context(context.Background())
+
+			changes, err := s.readChanges(ctx)
 			if err != nil {
-				if errors.Is(err, errRetryable) {
```

Comment:
Why aren't we checking this any more?

Reply:
Because it's never returned. The txn package now takes care of retries and was a vestigial of the last implementation. This means that adaptive polling might be harder...